ui/app/components/app/whats-new-popup/whats-new-popup.js

```diff
@@ -0,0 +1,190 @@
+        <Button
+          type="secondary"
+          className="whats-new-popup__button"
+          rounded
+          onClick={actionFunction}
+        >
+          {notification.actionText}
+        </Button>
+      )}
+    </div>
+  );
+};
+
+const renderSubsequentNotification = ({
+  notification,
+  id,
+  date,
+  index,
+  idRefMap,
+}) => {
+  const actionFunction = getActionFunctionById(id);
+  return (
+    <div
+      className={classnames('whats-new-popup__notification')}
+      key={`whats-new-popop-notificatiion-${index}`}
+      ref={idRefMap[id]}
+    >
+      <div className="whats-new-popup__notification-title">
+        {notification.title}
+      </div>
+      <div className="whats-new-popup__description-and-date">
+        <div className="whats-new-popup__notification-description">
+          {notification.description}
+        </div>
+        <div className="whats-new-popup__notification-date">{date}</div>
+      </div>
+      {notification.actionText && (
+        <div className="whats-new-popup__link" onClick={actionFunction}>
+          {notification.actionText}
+        </div>
+      )}
+    </div>
+  );
+};
+
+export default function WhatsNewPopup({ onClose }) {
+  const t = useContext(I18nContext);
+
+  const notifications = useSelector(getSortedNotificationsToShow);
+
+  const [seenNotifications, setSeenNotifications] = useState({});
+
+  const popoverRef = useRef();
+
+  const memoizedNotifications = useEqualityCheck(notifications);
+  const idRefMap = useMemo(
+    () =>
+      memoizedNotifications.reduce(
+        (_idRefMap, notification) => ({
+          ..._idRefMap,
+          [notification.id]: React.createRef(),
+        }),
+        {},
+      ),
+    [memoizedNotifications],
+  );
+
+  useEffect(() => {
+    const observer = new window.IntersectionObserver(
+      (entries, _observer) => {
+        entries.forEach((entry) => {
+          if (entry.isIntersecting) {
+            const [id, ref] = Object.entries(idRefMap).find(([_, _ref]) =>
+              _ref.current.isSameNode(entry.target),
+            );
+
+            setSeenNotifications((_seenNotifications) => ({
+              ..._seenNotifications,
+              [id]: true,
+            }));
+
+            _observer.unobserve(ref.current);
+          }
+        });
+      },
+      {
+        root: popoverRef.current,
+        threshold: 1.0,
+      },
+    );
+
+    Object.values(idRefMap).forEach((ref) => {
+      observer.observe(ref.current);
+    });
+
+    return () => {
+      observer.disconnect();
+    };
+  }, [idRefMap, setSeenNotifications]);
+
+  return (
+    <Popover
+      className="whats-new-popup__popover"
+      title={t('whatsNew')}
+      onClose={() => {
+        updateViewedNotifications(seenNotifications);
+        onClose();
+      }}
+      popoverRef={popoverRef}
+      mediumHeight
+    >
+      <div className="whats-new-popup__notifications">
+        {notifications.map(({ id, date }, index) => {
```

Comment:
Nit: I think the `date` here can be removed, both in this function and as a parameter to each of the notification rendering functions. It's in the `notification` object that is also passed into both functions.